### PR TITLE
Add `OUResolverExecutor`

### PR DESCRIPTION
**Pull request by thiva-k**
This pull request introduces a new executor, `OUResolverExecutor`, to the user onboarding flow, allowing users to be created in the admin's organization unit (OU) when specified. The changes include the implementation of the executor, its registration, usage in the onboarding flow, and comprehensive unit tests.

**Key changes:**

**User onboarding flow enhancement:**
* Added a new `ou_resolver` task to the `user_onboarding_flow.json` that runs after user type resolution and before email prompt, enabling conditional assignment of the user's OU to the admin's OU based on the `createInAdminOU` property.

**Executor implementation and registration:**
* Implemented the `OUResolverExecutor` in `ou_resolver_executor.go`, which checks the `createInAdminOU` node property and, if true, sets the user's OU to the admin's OU from the security context. Handles missing admin OU and property type errors gracefully.
* Registered the new executor in the executor initialization logic, and defined its name constant. [[1]](diffhunk://#diff-0fea251d329deae0b6b8ddab73d999044a47de1d1fc219263dcb7b1125885e06R92) [[2]](diffhunk://#diff-82a5330129814e06de3d54d982ddd0ee9e70262be578aa6dcfbe0babfa628d4fR45)

**Configuration and constants:**
* Added the `createInAdminOU` node property constant to the common flow constants.

**Testing:**
* Added thorough unit tests for `OUResolverExecutor`, covering scenarios such as successful OU override, missing admin OU, property absence, wrong property types, and nil contexts.

## Summary by CodeRabbit

* **New Features**
  * Added organizational unit (OU) resolution for workflows: can auto-resolve and override OU from the caller when configured, with explicit handling for unsupported strategies and skip behavior.

* **Tests**
  * Added comprehensive test coverage for the OU resolver covering success, missing caller OU, unsupported strategies, missing/wrong properties, nil context, and other edge cases.

### CODE REVIEW

File: backend/cmd/server/bootstrap/flows/user_onboarding/user_onboarding_flow.json [hunk 1]
```diff
@@ -27,9 +27,20 @@
...
+            "id": "ou_resolver",
+            "type": "TASK_EXECUTION",
+            "properties": {
+                "createInAdminOU": true
+            },
+            "executor": {
+                "name": "OUResolverExecutor"
+            },
+            "onSuccess": "prompt_email"
+        },
```
[the remark this answers is not in the document]
Comment (thiva-k) [reply backend/cmd/server/bootstrap/flows/user_onboarding/user_onboarding_flow.json]: This is fine for the initial implementation

File: backend/internal/flow/executor/ou_resolver_executor.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		logger:            logger,
+	}
+}
+
+// Execute resolves the organization unit for the user being onboarded.
+// If the createInAdminOU node property is true, it sets the ouId runtime data
+// to the admin's organization unit, overriding the default OU from the user type.
+func (e *ouResolverExecutor) Execute(ctx *core.NodeContext) (*common.ExecutorResponse, error) {
+	logger := e.logger.With(log.String(log.LoggerKeyFlowID, ctx.FlowID))
+
```
[the remark this answers is not in the document]
Comment (thiva-k) [reply backend/internal/flow/executor/ou_resolver_executor.go:60]: This is fine. ctx will never be nil here

File: backend/cmd/server/bootstrap/flows/user_onboarding/user_onboarding_flow.json (hunk 1 above)
Comment (senthalan) [inline backend/cmd/server/bootstrap/flows/user_onboarding/user_onboarding_flow.json]: let's not change the default behaviour. let's keep it to create the user in the selected user schema OU

File: backend/internal/flow/common/constants.go [hunk 3]
```diff
@@ -143,6 +143,8 @@ const (
 	NodePropertyAllowAuthenticationWithoutLocalUser = "allowAuthenticationWithoutLocalUser"
 	// NodePropertyAllowRegistrationWithExistingUser indicates whether registration is allowed with an existing user
 	NodePropertyAllowRegistrationWithExistingUser = "allowRegistrationWithExistingUser"
+	// NodePropertyCreateInAdminOU indicates whether to create the user in the admin's organization unit
+	NodePropertyCreateInAdminOU = "createInAdminOU"
```
Comment (senthalan) [inline backend/internal/flow/common/constants.go]: Shall we keep as `resolveFrom` configuration?

possible `resolveFrom` can be,

1. userType (Try to resolve the OUs where the users can be created with the selected user type, if there is only one, it will use it or prompt for user selection)
2. caller (The callers OU will be used when creating the user)

We can implement the caller option as of now, and improve to support userType later

File: backend/internal/flow/executor/ou_resolver_executor.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	}
+
+	// Check if the node property requests creating the user in the admin's OU.
+	if !e.shouldCreateInAdminOU(ctx) {
+		logger.Debug("createInAdminOU not enabled, skipping OU override")
+		return execResp, nil
+	}
+
+	// Extract the admin's OU from the security context.
+	adminOUID := security.GetOUID(ctx.Context)
```
Comment (senthalan) [inline backend/internal/flow/executor/ou_resolver_executor.go]: let's use the world caller here rather than admin. 
```suggestion
	// Extract the callers's OU from the security context.
	callerOUID := security.GetOUID(ctx.Context)
```

File: backend/cmd/server/bootstrap/flows/user_onboarding/user_onboarding_flow.json (hunk 1 above)
Comment (thiva-k) [reply backend/cmd/server/bootstrap/flows/user_onboarding/user_onboarding_flow.json]: Removed

File: backend/internal/flow/common/constants.go (hunk 3 above)
Comment (thiva-k) [reply backend/internal/flow/common/constants.go]: Added the caller implementation for now

File: backend/internal/flow/executor/ou_resolver_executor.go (hunk 4 above)
Comment (thiva-k) [reply backend/internal/flow/executor/ou_resolver_executor.go]: updated
